app/cdap/components/common/DataQualityCircularProgressBar/utils.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import {
+  IHeaderNamesList,
+  IDataQualityItem,
+} from 'components/WranglerGrid/SelectColumnPanel/types';
+import { IStatistics } from 'components/GridTable/types';
+
+export const getDataQuality = (statistics: IStatistics, columnList: IHeaderNamesList[]) => {
+  const dataQuality: IDataQualityItem[] = [];
+  columnList?.forEach((columnName: IHeaderNamesList) => {
```

Comment:
why isn't this a map? Also why is the optional chaining necessary if it is required in the type definition?

Reply:
we are making data format according to the need of UI that's why used forEach and storing values in array by pushing item to it. Removed optional chaining